### DISCUSSION TITLE
Issue label lozenge colours changed

### DISCUSSION THREAD

**Discussion by flaix** [community/community#22195] [Accessibility]
<p>I noticed the colours for issue labels changed. Is this a side effect of the introduction of dark mode? The colours changed for light mode, too.<br>
The problem is that text is not readable on lozenges with darker colours anymore.</p>

**Comment by gaknoll** [comment]
<p><a class="mention" href="/flaix">@flaix</a> are you using any browser extensions for GitHub? We’ve seen some issues with customers using the <code>refined-github</code> extension for example. The colors should not have changed for light mode, though due to how we dynamically calculate contrast for alpha in dark mode the colors could look visually different when in dark mode. If you’re not using an extension can you share a screenshot so we can investigate? Thanks!</p>

**Comment by bombasticbob** [comment]
<p>I am not using any extensions for github, simple firefox browser as always.  I’m not using any theme, either, just the default I’ve been using for several years.</p>
<p>The problem I am seeing is that the text color is ALWAYS BLACK.  It used to be that if I chose a color wth low luminocity, the text would change to WHITE.  This made it readable.  But for the vast majority of colors, black text will NOT work, especially when people NEED HIGH CONTRAST.  You know, like people with OLD eyes.</p>
<p>It’s actually an “accessibility” thing.</p>
<p>All we need is one of two things:<br>
a) fix it so it works the way it used to (text automatically becomes white when dark luminocity colors are used)<br>
b) let us change the text color for a label as well as the background [even better].</p>
<p>In any case I’ll end up re-doing EVERY label, including the default ones, because of this change in behavior.  Not what I wanted to spend time on, ya know?</p>

**Comment by gaknoll** [comment]
<p>Sorry you’re also experiencing this <a class="mention" href="/bombasticbob">@bombasticbob</a>! I’ve had to go through and redo all the label colors for a repo before (unrelated to dark mode :wink:) so I 100% feel you on not wanting to spend your time fixing this…<br>
But this should definitely not be happening and seems to be a weird bug affecting some users. I just shared this with the team to review</p>

**Comment by flaix** [comment]
<p>I am not using any browser extension either. The problem is, as bombasticbod described, that the text is always dark grey. It actually looks like the colours also changed, but that may be just a visual effect of the text color change and the added frame.</p>
<p>Now:<br>
<img src="https://user-images.githubusercontent.com/2362065/181069434-47bc3cf7-3d90-4bd6-8841-64b2f9c8691d.png" alt="GitHub-LabelColor-new" data-base62-sha1="eeYETGLphiqOGKe8hQ3DF94uvhI" width="211" height="43"></p>
<p>This is with Safari on macOS Mojave. I just opened an issues page with Firefox, and there the text is still white.</p>

**Comment by flaix** [comment]
<p>This is what it looked like before:</p>
<p><img src="https://user-images.githubusercontent.com/2362065/181069437-550872fb-1ff4-4635-9c72-21a220d6494a.png" alt="GitHub-LabelColor-old" data-base62-sha1="agZaYHQmqbxlP0nSJYAH0mTRUyP" width="211" height="43"></p>
<p>I had to put it into a new post, due to forum restriction.</p>

**Comment by jonrohan** [comment]
<p>Thank you <a class="mention" href="/flaix">@flaix</a> and <a class="mention" href="/bombasticbob">@bombasticbob</a> for the report. Could you provide specifics on browser versions and os versions?</p>

**Comment by flaix** [comment]
<p>macOS Mojave 10.14.6<br>
Safari 13.0.5 has dark grey text and border around lozenge<br>
Firefox 83.0 has white text and no border</p>

**Comment by jonrohan** [comment, marked as answer]
<p>Ah, ok. We were using a bit of CSS that Safari doesn’t get support for until 13.1. I have a fix coming that should fix for Safari, but won’t be able to deploy it until our code freeze is over next week.</p>
<p>To get a fix faster, If upgrading your browser is an option, I would suggest upgrading to at least Safari 13.1.</p>

**Comment by flaix** [comment]
<p>Ok, cool. I could wait for a fix, since you already have one. The only thing keeping me from updating Safari is the amount of open browser tabs I’d lose. :slight_smile: But I guess it is high time to upgrade to version 14, in general, so maybe this will be reason to finally do that.<br>
Thanks.</p>

**Comment by jonrohan** [comment]
<aside class="quote no-group" data-username="flaix" data-post="10" data-topic="148392">
<div class="title">
<div class="quote-controls"></div>
 flaix:</div>
<blockquote>
<p>The only thing keeping me from updating Safari is the amount of open browser tabs I’d lose.</p>
</blockquote>
</aside>
<p>:joy: Totally valid reason. From one tab addict to another, this History option has saved me a few times when my browser crashed. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/2/02918bddbae76b43014a47216f19a0a674e4ae7b.jpeg" data-download-href="/uploads/short-url/mIMLxLCXBb3sqY3Q7PiZQdQVrt.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/54012/181069443-e6b31ac9-1e29-499e-a44c-e5f9d2451e87.jpeg" alt="image" data-base62-sha1="mIMLxLCXBb3sqY3Q7PiZQdQVrt" width="690" height="284" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02918bddbae76b43014a47216f19a0a674e4ae7b_2_690x284.jpeg, https://aws1.discourse-cdn.com/github/original/3X/0/2/02918bddbae76b43014a47216f19a0a674e4ae7b.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/2/02918bddbae76b43014a47216f19a0a674e4ae7b.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02918bddbae76b43014a47216f19a0a674e4ae7b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1004×414 55.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by flaix** [comment]
<p>Hm, ok, ok, I’ll give it a shot, either later or tomorrow, when GitHub Universe is over. :slightly_smiling_face: Thanks!</p>

**Comment by bombasticbob** [comment]
<p>I’m using FreeBSD 12-STABLE and Firefox 70.0.1 on this machine.  (keep in mind that updating things on FreeBSD can be a non-trivial process, though it is AWESOME as a development platform and stability, over bleeding-edge, is generally a VERY good thing).  This version of firefox is from the 2019Q4 ports collection as I recall.</p>
<p>I also tested it with chromium version 78.0.3904.87 .  Same problem.</p>
<p>On a hunch I tested it using firefox 78 on the latest Devuan.  Seems to work as expected there, i.e. white text on a dark red color for labels, black for light colored labels.</p>
<p>uname says this: 4.19.0-12-amd64 <span class="hashtag">#1</span> SMP Debian 4.19.152-1 (2020-10-18) x86_64 GNU/Linux</p>
<p>Apparently the style sheet changes aren’t compatible with 1-2 year old firefox code.  That’s more than a bit disturbing, actually…</p>
<p>Just to mention, on very stable “I am trying to get work done” systems, it is common NOT to constantly update them.  this is especially true with servers, which may get security updates, but very few software version updates.  Additionally many Linux distributions will stabilize software versions at a specific point.  Debian is infamous for this, sometimes being 2 or 3 years behind ‘bleeding edge’ but it’s very very stable.</p>
<p>So I would ask very nicely that these changes  be tested on older browsers, please, especially if they’re LTS versions or are shipped as part of Linux distributions that have LTS support.</p>

**Comment by flaix** [comment]
<p>I upgraded to Safari 14.0.1, and at least for me that fixed the issue. Text colors are now back to normal.</p>

**Comment by cehfisher** [comment]
Hi @flaix @queenofcorgis - this issue seems to have been resolved [per the comment from the original poster](https://github.com/orgs/community/discussions/22195#discussioncomment-3235840). Could you please close as resolved?
